spring-content-rest/src/main/java/internal/org/springframework/content/rest/links/ContentLinksResourceProcessor.java

```diff
@@ -76,9 +76,10 @@ public PersistentEntityResource process(final PersistentEntityResource resource)
 
 		Object entityId = DomainObjectUtils.getId(object);
```

Comment:
Is that under the assumption that the projection actually exposes the actual id ? Or does that not matter ?

Reply:
Just FYI - I was using this in my own proof-of-concept (but that might be totally wrong):
```
Object entityId = DomainObjectUtils.getId(resource.getTarget());
```

Reply:
Yes, at the moment this assumes the projection exposes a method annotated with @Id that returns the id.  

I want to try and remove, if possible.

Reply:
I refactored this change a little bit and it now should handle projections correctly essentially the same as regular entities.  

Behind the scenes spring data is creating a proxy and with a bit of digging I discovered its possible to de-reference it to find the actual object.  Test proves you now don't need an annotated `getId` method.

You should be able to test this by grabbing the latest snapshots.  Available from `https://oss.sonatype.org/content/repositories/snapshots`.